device/transport/amqp/src/amqp.ts

```diff
@@ -92,7 +92,7 @@ export class Amqp extends EventEmitter implements Client.Transport {
     this._amqp = baseClient || new BaseAmqpClient(false, 'azure-iot-device/' + packageJson.version);
     this._amqp.setDisconnectHandler((err) => {
       debug('disconnected event handler: ' + (err ? err.toString() : 'no error'));
-      this._fsm.handle('disconnect', () => {
+      this._fsm.handle('disconnectError', err, () => {
```

Comment:
>disconnectError [](start = 24, length = 15)

I'm struggling with this name.  Without thinking too deeply, my first reaction is that this is handling an error that occurred while disconnecting.  I _think_ you mean that it's handling a disconnect that happened because of an error, but that doesn't come though.  I can't come up with a good name off the top of my head, but I think you want an action called something like "serverInitiatedDisconnect," but I know that this description doesn't cover all cases.  Maybe "conectionDropped" or "connectionSevered" to indicate that the disconnect wasn't user initiated.  Or maybe just "nonUserInitiatedDisconnect"?

And if this handler could cover both cases (user-initiated and non-user-initiated), then I need to understand more and maybe we want to separate those two cases into two different handlers.

Reply:
amqpConnectionClosed?


---
In reply to: [175762418](https://github.com/Azure/azure-iot-sdk-node/pull/254#discussion_r175762418) [](ancestors = 175762418)